httpx/_transports/mounts.py

```diff
@@ -0,0 +1,218 @@
+"""
+This module exposes the `httpx.Mount` and `httpx.AsyncMount` transport classes.
+"""
+
+import ipaddress
+import re
+import typing
+
+from .._exceptions import RequestError
+from .._models import Request, Response
+from .._urls import URL
+from .base import AsyncBaseTransport, BaseTransport
+
+
+class MountNotFound(RequestError):
+    pass
+
+
+class URLPattern:
```

Comment:
why duplicating it from `httpx._utils`?

Reply:
Well caught... because this pull request *will* also include removing `mounts=...` from the client, but I haven't gotten to that part yet.

It'll make more sense for all the implementation to be together in this module once the functionality is fully encompassed by `httpx.Mounts()`.